pybaseball/playerid_lookup.py

```diff
@@ -47,77 +50,116 @@ def chadwick_register(save: bool = False) -> pd.DataFrame:
 
 def get_lookup_table(save=False):
     table = chadwick_register(save)
-    #make these lowercase to avoid capitalization mistakes when searching
+    # make these lowercase to avoid capitalization mistakes when searching
     table['name_last'] = table['name_last'].str.lower()
     table['name_first'] = table['name_first'].str.lower()
     return table
 
 
-def playerid_lookup(last=None, first=None, player_list=None):
-    # force input strings to lowercase
-    if last:
-        last = last.lower()
-    if first:
-        first = first.lower()
-    table = get_lookup_table()
-
-    # if player_list has a value, then the user is passing in a list of players
-    # the list of players may be comma delimited for last, first, or just last
-    if player_list:
-        player_counter = 1
-        for player in player_list:
-            last = player.split(",")[0].strip()
-            first = None
-            if (len(player.split(",")) > 1):
-                first = player.split(",")[1].strip()
-            if (player_counter == 1):
-                results = playerid_lookup(last, first)
-            else:
-                results = results.append(playerid_lookup(last, first), ignore_index=True)
-            player_counter += 1
+def name_similarity(last: str, first: str, player_table: pd.DataFrame) -> pd.DataFrame:
+    """Calculates similarity of first and last name provided with all players in player_table
+
+    Args:
+        last (str): Provided last name
+        first (str): Provided first name
+        player_table (pd.DataFrame): Chadwick player table including names
+
+    Returns:
+        pd.DataFrame: 5 nearest matches from fuzzywuzzy.process
+    """
+    filled_df = player_table.fillna("")
+    chadwick_names = filled_df["name_first"] + " " + filled_df["name_last"]
+    fuzzy_matches = pd.DataFrame(process.extract(f"{first} {last}", chadwick_names, limit=5))
+    matched_names = fuzzy_matches[0].str.split(expand=True)
+    matched_names = matched_names.rename(columns={0: "name_first", 1: "name_last"})
+    return matched_names
+
+
+class player_search_client:
+    def __init__(self) -> None:
+        self.table = get_lookup_table()
```

Comment:
I think it'd be better to initialize this to `None`, then load it lazily the first time the `search` method gets called. 
example, 

```
class player_search_client:
  def __init__(self):
    self._table = None
  @property
  def table(self):
    if self._table is None:
      self._table = get_lookup_table()
   return self._table
```

then in `search` you can reference `self.table` as before